plane/src/proxy/proxy_service.rs

```diff
@@ -143,6 +143,14 @@ impl RequestHandler {
                 .body(hyper::Body::empty())?);
         };
 
+        // If the host header doesn't parse (non-ascii) or doesn't match the backend's stored subdomain, return an error
+        if !matches!(request_rewriter.get_subdomain(), Ok(request_subdomain) if request_subdomain == route_info.subdomain.as_deref())
```

Comment:
I find this a bit hard to parse, but if we land #684 first we can do something like:

```rust
let request_subdomain = request_rewriter.get_subdomain().map_err(|_| ProxyError::InvalidSubdomain)?;
if request_subdomain != route_info.subdomain.as_deref() {
    return Err(ProxyError::InvalidSubdomain);
}
```